mod/research/set/self/research_page/select_fields.rb

```diff
@@ -0,0 +1,49 @@
+format :html do
+  def metric_select
+    research_select_tag :metric, metric_list, metric
+  end
+
+  def year_select
+    research_select_tag :year, year_list, year
+  end
+
+  def research_select_tag name, items, selected
+    select_tag_with_html_options(
+      name, items, selected: selected, url: ->(item) { research_url(name => item) }
+    ) do |item, option_id, selected_option_id|
+      option =
+        wrap_with :div, id: option_id do
+          haml_partial "#{name}_option", item: item
```

Comment:
are these partials used elsewhere?  if not, why not put the wrap in there?

Reply:
to reduce duplication and complication. That method is called for several partials. If I move the wrap in the partial the I have to put in all partials and pass an additional argument to the partials. Additionally, when you create a new partial for a select tag option item you have to know that you have wrap it with this to make it work. The way it is now you can use any partial.